pinot-controller/src/main/java/org/apache/pinot/controller/helix/SegmentStatusChecker.java

```diff
@@ -135,6 +138,17 @@ protected void postprocess(Context context) {
     _controllerMetrics.setValueOfGlobalGauge(ControllerGauge.UPSERT_TABLE_COUNT, context._upsertTableCount);
     _controllerMetrics.setValueOfGlobalGauge(ControllerGauge.DISABLED_TABLE_COUNT, context._disabledTables.size());
 
+    _tierBackendGauges.forEach(_controllerMetrics::removeGauge);
```

Comment:
Do you want to clear `_tierBackendGauges` as well?
Or you can keep track all the valid metric names from this loop in a set: `context._tierBackendTableCountMap.forEach((tier, count) -> {...}` 
say `newTierBackendGauges`, then call
`_tierBackendGauges. retainAll(newTierBackendGauges)` to get the gauge names not existed.
Then call `removeGauge` and last thing is 
```
_tierBackendGauges.clear(); 
_tierBackendGauges.putAll(newTierBackendGauges)
```

Reply:
then you need to add `_tierBackendGauges.clear()` to remove all the existing gauges right?

Reply:
Yeah but there will be 1 entry per tier backend plugin which won't be much so I thought we can avoid the complication of managing the previous/new diff and just delete all gauges for the tiers which the controller has witnessed since boot up. What you think?